config/config.go

```diff
@@ -65,48 +60,74 @@ func create(configPath string) error {
 	return nil
 }
 
-// New returns parsed config
+func newEntry() *Entry {
+	return &Entry{
+		Launcher: Launcher{
+			Version: "stable",
+			Image:   "screwdrivercd/launcher",
+		},
+	}
+}
+
 func New(configPath string) (Config, error) {
 	err := create(configPath)
 	if err != nil {
 		return Config{}, err
 	}
 
-	configList, err := newConfigList(configPath)
+	file, err := os.Open(configPath)
 	if err != nil {
-		return Config{}, err
+		return Config{}, fmt.Errorf("failed to read config file: %v", err)
 	}
 
-	currentConfig, exists := configList.Configs[configList.Current]
-	if !exists {
-		return Config{}, fmt.Errorf("config `%s` does not exist", configList.Current)
+	var c = Config{
+		filePath: configPath,
 	}
 
-	currentConfig.filePath = configPath
+	err = yaml.NewDecoder(file).Decode(&c)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to parse config file: %v", err)
+	}
 
-	return currentConfig, nil
+	return c, nil
 }
 
-func newConfigList(configPath string) (configList, error) {
-	file, err := os.Open(configPath)
-	if err != nil {
-		return configList{}, fmt.Errorf("failed to read config file: %v", err)
+func (c *Config) Add(name string) error {
+	_, exist := c.Entries[name]
+	if exist {
+		return fmt.Errorf("config `%s` already exists", name)
 	}
-	defer file.Close()
 
-	var c = configList{}
+	c.Entries[name] = newEntry()
+	return nil
+}
+
+func (c *Config) Get(name string) (*Entry, error) {
+	currentEntry, exists := c.Entries[name]
+	if !exists {
+		return &Entry{}, fmt.Errorf("config `%s` does not exist", name)
+	}
 
-	err = yaml.NewDecoder(file).Decode(&c)
+	return currentEntry, nil
+}
 
+func (c *Config) Save() error {
+	file, err := os.OpenFile(c.filePath, os.O_RDWR|os.O_TRUNC, 0666)
 	if err != nil {
-		return configList{}, fmt.Errorf("failed to parse config file: %v", err)
+		return err
 	}
+	defer file.Close()
 
-	return c, nil
+	err = yaml.NewEncoder(file).Encode(c)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // Set preserve sd-local config with new value.
-func (c *Config) Set(key, value string) error {
+func (c *Entry) Set(key, value string) error {
```

Comment:
I think `e` is a good receiver name.

Reply:
I fixed that in https://github.com/screwdriver-cd/sd-local/pull/31/commits/44e2d54edbe88c5288bc934ae231ff428e2e4a31.